chatterbot/trainers.py

```diff
@@ -78,17 +77,25 @@ def train(self, conversation):
         Train the chat bot based on the provided list of
         statements that represents a single conversation.
         """
-        previous_statement_text = None
 
-        for text in conversation:
-            statement = self.get_or_create(text)
+        # Inline method yields statements for larger datasets
+        def get_statements(conversations):
+            previous_statement_text = None
+
+            for text in conversation:
+                statement = self.get_or_create(text)
+
+                if previous_statement_text:
+                    statement.add_response(
+                        Response(previous_statement_text)
+                    )
+
+                previous_statement_text = statement.text
+                yield statement
```

Comment:
Was there a performance increase after this change?
These changes are quite different from what was discussed in #815.
Generator functions are a bit different from list comprehension as they produce a "lazy list" that can be called later. I'm not sure if this actually changes anything since we are calling the generator right away in this function.